keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/probabilistic/TensorGammaVertex.java

```diff
@@ -0,0 +1,145 @@
+        this(getShapeProposal(a.getValue(), theta.getValue(), k.getValue()), a, theta, k, new KeanuRandom());
+    }
+
+    public TensorGammaVertex(DoubleTensorVertex a, DoubleTensorVertex theta, double k) {
+        this(a, theta, new ConstantTensorVertex(k), new KeanuRandom());
+    }
+
+    public TensorGammaVertex(DoubleTensorVertex a, double theta, DoubleTensorVertex k) {
+        this(a, new ConstantTensorVertex(theta), k, new KeanuRandom());
+    }
+
+    public TensorGammaVertex(DoubleTensorVertex a, double theta, double k) {
+        this(a, new ConstantTensorVertex(theta), new ConstantTensorVertex(k), new KeanuRandom());
+    }
+
+    public TensorGammaVertex(double a, DoubleTensorVertex theta, DoubleTensorVertex k) {
+        this(new ConstantTensorVertex(a), theta, k, new KeanuRandom());
+    }
+
+    public TensorGammaVertex(double a, DoubleTensorVertex theta, double k) {
+        this(new ConstantTensorVertex(a), theta, new ConstantTensorVertex(k), new KeanuRandom());
+    }
+
+    public TensorGammaVertex(double a, double theta, double k) {
+        this(new ConstantTensorVertex(a), new ConstantTensorVertex(theta), new ConstantTensorVertex(k), new KeanuRandom());
+    }
+
+    @Override
+    public double logPdf(DoubleTensor value) {
+        DoubleTensor aValues = a.getValue();
+        DoubleTensor thetaValues = theta.getValue();
+        DoubleTensor kValues = k.getValue();
+
+        DoubleTensor logPdfs = TensorGamma.logPdf(aValues, thetaValues, kValues, value);
+        return logPdfs.sum();
+    }
+
+    @Override
+    public Map<Long, DoubleTensor> dLogPdf(DoubleTensor value) {
+        TensorGamma.Diff dlnP = TensorGamma.dlnPdf(a.getValue(), theta.getValue(), k.getValue(), value);
+
+        return convertDualNumbersToDiff(dlnP.dPda, dlnP.dPdtheta, dlnP.dPdk, dlnP.dPdx);
+    }
+
+    private Map<Long, DoubleTensor> convertDualNumbersToDiff(DoubleTensor dPda,
+                                                               DoubleTensor dPdtheta,
+                                                               DoubleTensor dPdk,
+                                                               DoubleTensor dPdx) {
+
+        TensorPartialDerivatives dPdInputsFromA = a.getDualNumber().getPartialDerivatives().multiplyBy(dPda);
+        TensorPartialDerivatives dPdInputsFromTheta = theta.getDualNumber().getPartialDerivatives().multiplyBy(dPdtheta);
+        TensorPartialDerivatives dPdInputsFromK = k.getDualNumber().getPartialDerivatives().multiplyBy(dPdk);
+        TensorPartialDerivatives dPdInputs = dPdInputsFromA.add(dPdInputsFromTheta).add(dPdInputsFromK);
+
+        if (!this.isObserved()) {
+            dPdInputs.putWithRespectTo(getId(), dPdx);
+        }
+
+        return dPdInputs.asMap();
+    }
+
+    @Override
+    public DoubleTensor sample() {
+        return TensorGamma.sample(getValue().getShape(), a.getValue(), theta.getValue(), k.getValue(), random);
```

Comment:
Is this getValue().getShape() going to cause problems? Should it be just getShape()?